File: frontend/angular/src/app/models/endereco.model.ts
import {Cliente} from './cliente.model';

export interface Endereco {
  id?: number;
  logradouro: string;
  numero: string;
  cep: string;
  bairro: string;
  telefone: string;
  cidade: string;
  estado: string;
  enderecoPrincipal: boolean;
  complemento: string;
  cliente: Cliente;
}

File: backend/springboot/src/main/java/com/desafio/model/EnderecoModel.java
package com.desafio.model;

@Entity
public class EnderecoModel {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank(message = "Logradouro é obrigatório")
    private String logradouro;

    @NotBlank(message = "Número é obrigatório")
    private String numero;

    @NotBlank(message = "CEP é obrigatório")
    private String cep;

    private String bairro;
    private String telefone;
    private String cidade;
    private String estado;

    private boolean enderecoPrincipal = false;
    private String complemento;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;

    // Getters e Setters
}